ResearchKit/+Medable/ORKFormItemCell+Medable.m

```diff
@@ -0,0 +1,95 @@
+//
+//  ORKFormItemCell+Medable.m
+//  Axon
+//
+//  Created by me on 5/18/18.
+//  Copyright © 2018 Medable Inc. All rights reserved.
+//
+
+#import "ResearchKit.h"
+#import "ORKFormItemCell.h"
+
+@interface ORKFormItemCell ()
+- (void)ork_setAnswer:(id)answer;
+@end
+
+#pragma mark -
+
+@implementation ORKFormItemTextFieldBasedCell (Medable)
+
+- (void)ork_setAnswer:(id)answer
+{
+    [super ork_setAnswer:answer];
+    
+    @try
+    {
+        // in code below, use KVC, and performSelector:withObject:
+        // in order to bypass compiler/linker restrictions and
+        // avoid adding explicit dependencies to this module (RK)
+        
+        Class DBZxcvbn = NSClassFromString(@"DBZxcvbn");
+        NSString* answerFormatKeyPath = @"answerFormat.";
+
+        NSString* isSecureTextEntryKeyPath =
+        [answerFormatKeyPath stringByAppendingString:@"isSecureTextEntry"];
+        
+        NSString* minPasswordStrengthKeyPath =
+        [answerFormatKeyPath stringByAppendingString:@"minimumPasswordStrength"];
+        
+        int minimumPasswordStrength =
+        [[self.formItem valueForKeyPath:minPasswordStrengthKeyPath] intValue];
+        
+        if (DBZxcvbn && minimumPasswordStrength &&
+            [[self.formItem valueForKeyPath:isSecureTextEntryKeyPath] boolValue])
+        {
+            enum { width = 30 };
+            BOOL answerIsEmpty = [answer isEqual:NSNull.null];
+            UITextField* textField = [self valueForKey:@"textField"];
+            UILabel* indicator = ((UILabel*)textField.leftView ?:
+                                  [[UILabel alloc] initWithFrame:
+                                   CGRectMake(0, 0, width, width)]);
+            
+            textField.leftView = indicator;
+            textField.leftViewMode = UITextFieldViewModeAlways;
+            
+            #pragma clang diagnostic push
+            #pragma clang diagnostic ignored "-Warc-performSelector-leaks"
+            id result = (!answerIsEmpty ?
+                         [[DBZxcvbn new] performSelector:
+                          NSSelectorFromString(@"passwordStrength:")
+                                              withObject:answer] : nil);
+            #pragma clang diagnostic pop
+
+            int score = [[result valueForKey:@"score"] intValue];
```

Comment:
Use NS integers.

Reply:
if DBZ implementation changes, an exception would fire, which is why all this code is wrapped in a try block, so it can gracefully degrade by simply doing nothing.